docs/README.md

```diff
@@ -0,0 +1,191 @@
+Type Node Observability Operator into the filter box and select it.
+4. Click Install
+
+### Deployment from a custom catalog
+
+1. Prepare an index image
+```sh
+export BUNDLE_IMG=<registry>/<username>/node-observability-operator-bundle:<version>
+export INDEX_IMG=<registry>/<username>/node-observability-operator-bundle-index:<version>
+make index-image-build index-image-push
+```
+2. Ensure image registry is accessible,
+   1. by declaring the necessary credentials as secrets,
+   ```sh
+   oc -n openshift-marketplace create secret generic nodeobs-olm-secret  --type=kubernetes.io/dockercfg  --from-file=.dockercfg=${XDG_RUNTIME_DIR}/containers/auth.json
+   oc -n openshift-marketplace secrets link default nodeobs-olm-secret --for=pull
+   ```
+   2. Adding an `ImageContentSourcePolicy` if needed
+   ```sh
+   cat <<EOF  | oc create -f -
+   apiVersion: operator.openshift.io/v1alpha1
+   kind: ImageContentSourcePolicy
+   metadata:
+     name: brew-registry
+   spec:
+     repositoryDigestMirrors:
+     - mirrors:
+       - brew.registry.redhat.io
+       source: registry.redhat.io
+     - mirrors:
+       - brew.registry.redhat.io
+       source: registry.stage.redhat.io
+     - mirrors:
+       - brew.registry.redhat.io
+       source: registry-proxy.engineering.redhat.com
+   EOF
+   ```
+3. Create a CatalogSource object:
+```sh
+cat <<EOF | oc apply -f - 
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: node-observability-operator
+  namespace: openshift-marketplace
+spec:
+  sourceType: grpc
+  image: ${INDEX_IMG}
+EOF
+```
+4. Follow same steps as deployment from public release (see above)
+
+## Prepare to run profiling queries
+
+In order to be able to run profiling queries on a subset of worker nodes, a `NodeObservability` custom resource needs to be created.
+
+The following `NodeObservability` will create a daemonset targetting all nodes with label `app: example`, running a `node-observability-agent` pod on each of those node using the image given in `.spec.image`.
+
+```yaml
+apiVersion: nodeobservability.olm.openshift.io/v1alpha1
+kind: NodeObservability
+metadata:
+  name: nodeobservability-sample
+  namespace: node-observability-operator
+spec:
+  labels:
+    "node-role.kubernetes.io/worker": ""
+  image: "brew.registry.redhat.io/rh-osbs/node-observability-agent:0.1.0-3"
+```
+
+The CRIO unix socket of the underlying node is mounted on the agent pod, thus allowing the agent to communicate with CRIO to run the pprof request.
+
+The `kubelet-serving-ca` certificate chain is also mounted on the agent pod, which allows secure communication between agent and node's kubelet endpoint.
+
+## Run profiling queries
+
+Profiling query is a blocking operation and contains about 30 seconds
+worth of profiling (kubelet + crio `/pprof`) data. As such, only one
+query can be requested concurrently.
+
+Profiling queries can be requested with creating a `NodeObservabilityRun`
```

Comment:
You can request a Profiling queries by creating a `NodeObservabilityRun` resource.